web-app/src/main/webapp/script.js

```diff
@@ -70,17 +75,64 @@ function showReportForm() {
 }
 
 async function loadPoliceReports(map) {
+  // Clear all markers on the map
+  for (const marker of mapMarkers) {
+    marker.setMap(null);
+  }
+  mapMarkers.length = 0;
+
   const FILE_NAMES = ['2019_12_london', '2020_01_london', '2020_02_london', '2020_03_london', '2020_04_london', '2020_05_london']
+  const uncheckedCategoriesElement = Array.from(document.querySelectorAll("input.category:not(:checked)"));
+  const uncheckedCategories = uncheckedCategoriesElement.map(element => element.value);
+  const numberofMonths = Number(document.querySelector("input.time-frame:checked").value);
+
   for (const file_name of FILE_NAMES) {
-    const data = await fetch('../data/' + file_name + '.json');
-    const reports = await data.json();
-    for (report of reports) {
-      new google.maps.Marker({
-        position: {
-          lat: Number(report.latitude),
-          lng: Number(report.longitude)
-        }, map: map
-      });
-    };
+    createPoliceReportMarkers(map, file_name, uncheckedCategories, numberofMonths);
   }
 }
+
+async function createPoliceReportMarkers(map, file_name, uncheckedCategories, numberofMonths) {
+  const data = await fetch('../data/' + file_name + '.json');
+  const reports = await data.json();
+  
+  if (reports.length != 0 && !isReportwithinTimeFrame(reports[0].month, numberofMonths)) {
```

Comment:
nit: you're passing `reports[0].month` here but `isReportwithinTimeFrame` takes both month and year?